spec/system/schools/users_spec.rb

```diff
@@ -346,6 +346,20 @@
           expect(other_school_admin.cluster_schools_for_switching).to eq([school])
         end
 
+        context 'when other user is staff' do
+          let!(:other_school_admin) { create(:staff, name: 'Other admin') }
+
+          it 'adds the user' do
+            click_on 'Add an existing Energy Sparks user as a school admin'
+            fill_in 'Email', with: other_school_admin.email
+            click_on 'Add user'
```

Comment:
I would split this into a before block and at least one test here as follows:

```
before do
  click_on 'Add an existing Energy Sparks user as a school admin'
  fill_in 'Email', with: other_school_admin.email
  click_on 'Add user'
end

it 'adds the user' do
  expect(page).to have_content(other_school_admin.name)
  other_school_admin.reload
  expect(other_school_admin.cluster_schools_for_switching).to eq([school])
  expect(other_school_admin.role).to eq 'school_admin'
end

```

Reply:
I've reworked the spec a bit to reduce overlap between the tests.